src/combinatorics/simple_graph/basic.lean

```diff
@@ -317,6 +317,30 @@ instance decidable_mem_edge_set [decidable_rel G.adj] :
 instance edges_fintype [decidable_eq V] [fintype V] [decidable_rel G.adj] :
   fintype G.edge_set := subtype.fintype _
 
+section from_edge_set
+
+variable (s : set (sym2 V))
+
+/--
+`from_edge_set` constructs a `simple_graph` from a set of edges, without loops.
+-/
+def from_edge_set : simple_graph V :=
+{ adj := λ v w, v ≠ w ∧ ⟦(v, w)⟧ ∈ s,
+  symm := λ v w h, ⟨h.1.symm, by {rw sym2.eq_swap, exact h.2}⟩ }
```

Comment:
We can make use of a `sym2` construction and the lattice structure on relations:
```
def from_edge_set : simple_graph V :=
{ adj := ne ⊓ sym2.to_rel s,
  symm := λ v w h, ⟨h.1.symm, sym2.to_rel_symmetric s h.2⟩ }
```
This change does not affect any of the following lemmas.

The name of the definition is ok, but I wonder if there is a better one. Maybe `of_edges` so you write `simple_graph.of_edges`? Not sure.

Reply:
The definition looks nicer, thanks!
Regarding the name, I thought of different variants. I wanted a distinction between `set` and `finset` constructions, so I decided to include `set/finset` in the name, drawing a parallel with already defined `edge_set/edge_finset`.

Reply:
The name is fine as it is, so I'll resolve this thread.